api/src/test/java/org/openmrs/module/bedmanagement/HibernateBedManagementDaoTest.java

```diff
@@ -430,4 +430,15 @@ public void shouldDeleteBedTag() throws Exception {
 		Assert.assertNull(bedTag);
 	}
 	
+	@Test
+	public void shouldDeleteBedLocationMapping() throws Exception {
+		Location location = locationDao.getLocationByUuid("98bc9b32-9d1a-11e2-8137-0800271c1b56");
+		List<BedLocationMapping> beforeDeleteBedLocationMappings = bedManagementDao
+		        .getBedLocationMappingByLocation(location);
+		
+		bedManagementDao.deleteBedLocationMapping(beforeDeleteBedLocationMappings.get(0));
+		List<BedLocationMapping> afterDeleteBedLocationMappings = bedManagementDao.getBedLocationMappingByLocation(location);
+		Assert.assertEquals(beforeDeleteBedLocationMappings.size() - 1, afterDeleteBedLocationMappings.size());
```

Comment:
Strictly speaking this is not sufficient, you should also check that the correct bed location mapping is gone after deletion.

Also this method should be renamed:
`getBedLocationMappingByLocation` => `getBedLocationMappingsByLocation`.
It returns a list so it's plural for mappings. If there is a similar method at the service layer then please rename it as well.

Reply:
@mks-d made changes.